1. Write a function that inputs a number and prints the multiplication table of that number


In [1]:
num=int(input("Enter any number: "))
table=[i*num for i in range(1,11,1)]
print("Table of {} is {} ".format(num,table))

Enter any number: 17
Table of 17 is [17, 34, 51, 68, 85, 102, 119, 136, 153, 170] 


2. Write a program to print twin primes less than 1000. If two consecutive odd numbers are
both prime then they are known as twin primes

In [2]:
import math 
arr=[1]*1000
arr[0]=0
arr[1]=0
# Here I am using Sieve of Eratosthenes
for i in range(int(math.sqrt(1000))):
  if arr[i]==1:
    j=2
    while i*j < 1000:
      arr[i*j]=0
      j=j+1
out=[]
for i in range(2,1000,1):
  if i%2!=0:
    if arr[i]==1 and arr[i+2] : 
      out.append(i)
      out.append(i+2)
print("Twin primes less than 1000 is {}".format(out))


Twin primes less than 1000 is [3, 5, 5, 7, 11, 13, 17, 19, 29, 31, 41, 43, 59, 61, 71, 73, 101, 103, 107, 109, 137, 139, 149, 151, 179, 181, 191, 193, 197, 199, 227, 229, 239, 241, 269, 271, 281, 283, 311, 313, 347, 349, 419, 421, 431, 433, 461, 463, 521, 523, 569, 571, 599, 601, 617, 619, 641, 643, 659, 661, 809, 811, 821, 823, 827, 829, 857, 859, 881, 883]


3. Write a program to find out the prime factors of a number. Example: prime factors of 56 -
2, 2, 2, 7

In [3]:
num = int(input("Enter the number whoes prime factorization you want"))
temp=num
l1=[]
for i in range(2,temp):
  if temp%i==0:
    while temp%i==0:
      l1.append(i)
      temp//=i
print("Prime factorization of {} is {}".format(num,l1))

Enter the number whoes prime factorization you want56
Prime factorization of 56 is [2, 2, 2, 7]


4. Write a program to implement these formulae of permutations and combinations.
Number of permutations of n objects taken r at a time: p(n, r) = n! / (n-r)!. Number of
combinations of n objects taken r at a time is: c(n, r) = n! / (r!*(n-r)!) = p(n,r) / r

In [0]:
def fact(n):
  """This function return factoriol of given number"""
  if n<=1:
    return 1
  else:
    return n*fact(n-1)

In [5]:
n=int(input("Enter the value of n:" ))
r=int(input("Enter the value of r:"))
# Calculation of p(n, r) = n! / (n-r)!.
result=fact(n)/fact(n-r)
print("The result of p(n, r) is {} ".format(result))
# Calculation of c(n, r) = n! / (r!*(n-r)!)
print("The result of c(n, r) is {} ".format(result/r))

Enter the value of n:5
Enter the value of r:4
The result of p(n, r) is 120.0 
The result of c(n, r) is 30.0 


5. Write a function that converts a decimal number to binary number

In [6]:
n = int(input("Enter the any number whoes binary conversion you want : "))
l1=[]
temp=n
while n != 0:
  l1.append(n%2)
  n=n//2
l1.reverse()
print("Decimal to Binary conversion of {} is{}: ".format(temp,l1))


Enter the any number whoes binary conversion you want : 23
Decimal to Binary conversion of 23 is[1, 0, 1, 1, 1]: 


6. Write a function cubesum() that accepts an integer and returns the sum of the cubes of
individual digits of that number. Use this function to make functions PrintArmstrong() and
isArmstrong() to print Armstrong numbers and to find whether is an Armstrong number

In [0]:
# function cubesomme() 
def cubesum(n):
  sum=0
  while n != 0:
    sum+=(n%10)**3
    n=n//10
  return sum
#function isArmstrong()
def isArmstrong(n):
  if(n==cubesum(n)):
    return True
  else:
    return False


In [8]:
#function printArmstrong()
num=int(input("Enter your number"))
if(isArmstrong(num)):
  print("The number {} is  Armstrong".format(num)) 
else:
  print("The number {} is  not an Armstrong".format(num))

Enter your number153
The number 153 is  Armstrong


7. Write a function prodDigits() that inputs a number and returns the product of digits of that
number.

In [0]:
def prodDigits(n):
  prod=1
  temp=int(n)
  while temp !=0:
    prod=prod*(temp%10)
    temp//=10
  return prod

In [10]:
num=int(input("Enter your number whoes product of digit you want"))
print("The Product of digit of {} is {} ".format(num,prodDigits(num)))

Enter your number whoes product of digit you want341
The Product of digit of 341 is 12 


8. If all digits of a number n are multiplied by each other repeating with the product, the one
digit number obtained at last is called the multiplicative digital root of n. The number of
times digits need to be multiplied to reach one digit is called the multiplicative
persistance of n.
Example: 86 -> 48 -> 32 -> 6 (MDR 6, MPersistence 3)
 341 -> 12->2 (MDR 2, MPersistence 2)
Using the function prodDigits() of previous exercise write functions MDR() and
MPersistence() that input a number and return its multiplicative digital root and
multiplicative persistence respectively

In [11]:
num=int(input("Enter any number you want to enter: "))
count=0
temp=num
while len(str(temp))>1:
  count+=1
  temp=prodDigits(temp)
print("MDR {}, MPersistence {}".format(temp,count))

Enter any number you want to enter: 86
MDR 6, MPersistence 3


9. Write a function sumPdivisors() that finds the sum of proper divisors of a number. Proper
divisors of a number are those numbers by which the number is divisible, except the
number itself. For example proper divisors of 36 are 1, 2, 3, 4, 6, 9, 18

In [0]:
def sumPdivisors(num):
  l1=[]
  l1.append(1)
  for i in range(2,int(math.sqrt(num)),1):
    if num%i==0:
      l1.append(i)
      l1.append(num//i)
  l1.sort()
  return l1
 

In [13]:
num = int (input("Enter your number Woes proper divesor you want:"   ))
print("Proper divisors of {} is {} ".format(num,sumPdivisors(num)))

Enter your number Woes proper divesor you want:36
Proper divisors of 36 is [1, 2, 3, 4, 9, 12, 18] 


10. A number is called perfect if the sum of proper divisors of that number is equal to the
number. For example 28 is perfect number, since 1+2+4+7+14=28. Write a program to
print all the perfect numbers in a given range

In [14]:
num=int(input("enter your range: "))
l1=[]
for n in range(1,num):
  sum=1
  for j in range(2,int(math.sqrt(n))+1,1):
    if n%j==0:
      sum+=j+n//j
  if sum ==n:
    l1.append(n)
print("The list of perfect sum in given range is {}".format(l1))



enter your range: 30
The list of perfect sum in given range is [1, 6, 28]


11. Two different numbers are called amicable numbers if the sum of the proper divisors of
each is equal to the other number. For example 220 and 284 are amicable numbers.Sum of proper divisors of 220 = 1+2+4+5+10+11+20+22+44+55+110 = 284
Sum of proper divisors of 284 = 1+2+4+71+142 = 220
Write a function to print pairs of amicable numbers in a range

In [0]:
def isAmicable(num1,num2):
  l2=sumPdivisors(num1)
  l3=sumPdivisors(num2)
  sum1,sum2=0,0
  for ele in l2:
    sum1+=ele
  for ele in l3:
    sum2+=ele
  if sum1==num2 and  sum2 == num1:
    print("the number {} and {} is amicable".format(num1,num2))
  else:
    print ("the number {} and {} is not amicable".format(num1,num2))

In [16]:
num1=int(input("Enter first number"))
num2=int(input("Enter second number"))
isAmicable(num1,num2)


Enter first number220
Enter second number284
the number 220 and 284 is amicable


12. Write a program which can filter odd numbers in a list by using filter function

In [17]:
#This block is used to take input in list 
num=int(input("Enter the no of element in list: "))
print("please enter {} number in list:".format(num))
l1=[]
for i in range(num):
  l1.append(int(input()))

Enter the no of element in list: 6
please enter 6 number in list:
45
3
67
2
9
31


In [21]:
l2=list(filter(lambda x:(x%2!=0),l1))
print("The odd number in your list are {}".format(l2))

The odd number in your list are [45, 3, 67, 9, 31]


13. Write a program which can map() to make a list whose elements are cube of elements in
a given list

In [22]:
l3=list(map(lambda x: (x**3),l1))
print("The cube of given list is {}".format(l3))

The cube of given list is [91125, 27, 300763, 8, 729, 29791]


14. Write a program which can map() and filter() to make a list whose elements are cube of
even number in a given list

In [23]:
l2=list(filter(lambda x:(x%2==0),l1))
l3=list(map(lambda x: (x**3),l2))
print("The list of elements are cube of even number in a given list {}".format(l3))

The list of elements are cube of even number in a given list [8]
